lib/iris/tests/unit/analysis/test_MAX_RUN.py

```diff
@@ -0,0 +1,300 @@
+                    [1, 1, 1, 3, 1, 3],
+                    [1, 1, 1, 3, 1, 3],
+                    [1, 1, 1, 3, 1, 3],
+                    [1, 1, 1, 3, 1, 3],
+                ]
+            ),
+            np.array(
+                [
+                    [0, 0, 0, 0, 0, 0],  # No mask
+                    [0, 0, 0, 0, 1, 1],  # Mask misses run
+                    [1, 1, 1, 0, 0, 0],  # Mask max run
+                    [0, 0, 1, 0, 0, 0],  # Partially mask run
+                    [1, 1, 1, 1, 1, 1],  # All mask
+                    [1, 1, 1, 1, 0, 1],  # All mask or run
+                ]
+            ),
+        ).T
+
+        self.expected_2d_axis0 = np.array([3, 3, 1, 2, 0, 1])
+
+        self.data_2d_axis1 = self.data_2d_axis0.T
+        self.expected_2d_axis1 = self.expected_2d_axis0
+
+
+class RealMixin:
+    def run_func(self, *args, **kwargs):
+        return MAX_RUN.call_func(*args, **kwargs)
+
+    def check_array(self, result, expected):
+        self.assertArrayEqual(result, expected)
+
+
+class LazyMixin:
+    def run_func(self, *args, **kwargs):
+        return MAX_RUN.lazy_func(*args, **kwargs)
+
+    def check_array(self, result, expected):
+        self.assertTrue(is_lazy_data(result))
+        result = as_concrete_data(result)
+        self.assertArrayEqual(result, expected)
+
+
+class TestBasic(UnmaskedTest, RealMixin):
+    def test_1d(self):
+        for data, expected in self.data_1ds:
+            result = self.run_func(
+                data,
+                axis=0,
+                function=bool_func,
+            )
+            self.check_array(result, expected)
+
+    def test_2d_axis0(self):
+        result = self.run_func(
+            self.data_2d_axis0,
+            axis=0,
+            function=bool_func,
+        )
+        self.check_array(result, self.expected_2d_axis0)
+
+    def test_2d_axis1(self):
+        result = self.run_func(
+            self.data_2d_axis1,
+            axis=1,
+            function=bool_func,
+        )
+        self.check_array(result, self.expected_2d_axis1)
+
+
+class TestLazy(UnmaskedTest, LazyMixin):
+    def test_1d(self):
+        for data, expected in self.data_1ds:
+            data = as_lazy_data(data)
+            result = self.run_func(
+                data,
+                axis=0,
+                function=bool_func,
+            )
+            self.check_array(result, expected)
+
+    def test_2d_axis0(self):
+        data = as_lazy_data(self.data_2d_axis0)
+        result = self.run_func(
+            data,
+            axis=0,
+            function=bool_func,
+        )
+        self.check_array(result, self.expected_2d_axis0)
+
+    def test_2d_axis1(self):
+        data = as_lazy_data(self.data_2d_axis1)
+        result = self.run_func(
+            data,
+            axis=1,
+            function=bool_func,
+        )
+        self.check_array(result, self.expected_2d_axis1)
+
+
+class TestLazyChunked(UnmaskedTest, LazyMixin):
```

Comment:
It might be worthwhile checking that the chunks in the returned array are as expected.

Reply:
Now done